src/components/product-card/ProductCard.jsx

```diff
@@ -1,36 +1,128 @@
 import { CartIcon, HeartIcon, Star } from "../../assets/icons";
 import { sold_out_img } from "../../assets/image";
+import { useCart } from "../../context/cart-context";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import { discount } from "../../utils/discount";
 import "./product-card.css";
 
 const ProductCard = ({ card }) => {
-   //calculating the discount percent of products
-   const discount = () => {
-      const discountPercent =
-         ((card.price - card.discountPrice) / card.price) * 100;
-      return Math.trunc(discountPercent);
+   const { state, dispatch } = useCart();
+
+   //adding a product to cart
+   const addToCart = async (card) => {
+      try {
+         const response = await axios.post(
+            "/api/user/cart",
+            {
+               product: card,
+            },
+            {
+               headers: {
+                  authorization: localStorage.getItem("token"),
```

Comment:
suggestion: you can get the data from localStorage once and reuse instead of getting it in every call.

Reply:
Thanks for your suggestion, will work on it.